dev/src/main/java/org/eclipse/codewind/intellij/module/CodewindModuleBuilder.java

```diff
@@ -0,0 +1,130 @@
+import com.intellij.openapi.module.*;
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.progress.EmptyProgressIndicator;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.roots.ui.configuration.ModulesProvider;
+import com.intellij.openapi.util.InvalidDataException;
+import org.eclipse.codewind.intellij.core.FileUtil;
+import org.eclipse.codewind.intellij.core.Logger;
+import org.eclipse.codewind.intellij.core.cli.InstallUtil;
+import org.eclipse.codewind.intellij.core.cli.ProjectUtil;
+import org.eclipse.codewind.intellij.core.connection.ProjectTemplateInfo;
+import org.jdom.JDOMException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeoutException;
+
+public class CodewindModuleBuilder extends JavaModuleBuilder {
+
+    private ProjectTemplateInfo template;
+
+    public CodewindModuleBuilder() {
+    }
+
+    @Override
+    public ModuleType getModuleType() {
+        return CodewindModuleType.CODEWIND_MODULE;
+    }
+
+    @Nullable
+    @Override
+    public ModuleWizardStep getCustomOptionsStep(WizardContext context, Disposable parentDisposable) {
+        try {
+            if (!InstallUtil.getInstallStatus().isInstalled()) {
+                return new InstallCodewindStep();
+            }
+            if (!InstallUtil.getInstallStatus().isStarted()) {
+                return new StartCodewindStep();
+            }
+            return createNewCodewindProjectStep();
+        } catch (Exception e) {
+            Logger.logWarning(e);
+        }
+        return null;
+    }
+
+    private ModuleWizardStep createNewCodewindProjectStep() throws JSONException, TimeoutException, IOException {
+        return new NewCodewindProjectStep(this);
+    }
+
+    @Override
+    public ModuleWizardStep[] createWizardSteps(@NotNull WizardContext wizardContext, @NotNull ModulesProvider modulesProvider) {
+        try {
+            if (!InstallUtil.getInstallStatus().isStarted())
+                return new ModuleWizardStep[]{createNewCodewindProjectStep()};
+        } catch (Exception e) {
+            Logger.logWarning(e);
+        }
+        return ModuleWizardStep.EMPTY_ARRAY;
+    }
+
+    @NotNull
+    @Override
+    public Module createModule(@NotNull ModifiableModuleModel moduleModel) throws InvalidDataException, IOException, ModuleWithNameAlreadyExists, JDOMException, ConfigurationException {
+        String path = getModuleFileDirectory();
+        String name = getName();
+        String url = template.getUrl();
+        String language = template.getLanguage();
+        String projectType = template.getProjectType();
+        String conid = "local";
+        try {
+            // Codewind won't create a project in a non-empty directory, and by this
```

Comment:
Is there something in Codewind that can be done?  eg. instead of non-empty directories, can they relax that restriction and at least ignore all .* files ?